### ISSUE TITLE
Text container index was invalid.

### ISSUE THREAD

**Issue by def-** [ddnet/ddnet#6640]
![image](https://github.com/ddnet/ddnet/assets/2335377/d1edc970-f21e-40f2-bd90-2df5021ed33d)

Reported by Nice on Discord:

> I was playing Copy Love Box. I had gotten guns and returned to spawn

@Robyt3 comes from https://github.com/ddnet/ddnet/commit/0c32eacbab28b8e2ef8b22d6fc82f4d00253c1fd I guess

**Comment by Jupeyy** [comment]
should be fixed in #6634

**Comment by Robyt3** [comment]
Can you ask them for the crash dump? With the back trace we can check where the invalid use is. Maybe it was fixed by #6634 already.

**Comment by def-** [comment]
```
-------------------

Error occurred on Sunday, May 21, 2023 at 13:29:53.

DDNet.exe caused an Illegal Instruction at location 00007FF6FC6A0F87 in module DDNet.exe.

AddrPC           Params
00007FF6FC6A0F87 0000003300000000 00000208FC2F4200 000000000000049D  DDNet.exe!0x300f87
00007FF6FC558EBB 0000000000000020 00000208FC2F0CF0 00000033B0DFAB90  DDNet.exe!0x1b8ebb
00007FF6FC63C812 0000000000000000 0000000000000000 0000000000000000  DDNet.exe!0x29c812
00007FF6FC63CCD7 00000000BF800000 00007FF6FC566112 0000000043888000  DDNet.exe!0x29ccd7
00007FF6FC40E9B7 0000000000000000 0000000000000000 0000000000000000  DDNet.exe!0x6e9b7
00007FF6FC40F08A 00007FF6FC6B63B8 00007FF6FC556036 0000000000000000  DDNet.exe!0x6f08a
00007FF6FC47DFE5 000249C300007F00 0000000000000000 0000000000000000  DDNet.exe!0xddfe5
00007FF6FC3BFBD1 00007FFC66FC8BF8 0000000000000000 00000033B0DFB1A0  DDNet.exe!0x1fbd1
00007FF6FC3C9348 00000033B0DFF4E0 00000033B0DFF500 00000033B0DFEE83  DDNet.exe!0x29348
00007FF6FC69F0AD 0000000000000001 00000208FC488210 00007FF6FC81CA08  DDNet.exe!0x2ff0ad
00007FF6FC3A13B1 0000000000000000 0000000000000000 0000000000000000  DDNet.exe!0x13b1
00007FF6FC3A14C6 0000000000000000 0000000000000000 0000000000000000  DDNet.exe!0x14c6
00007FFC6AB87614 0000000000000000 0000000000000000 0000000000000000  KERNEL32.DLL!BaseThreadInitThunk+0x14
00007FFC6B1426A1 0000000000000000 0000000000000000 0000000000000000  ntdll.dll!RtlUserThreadStart+0x21

00007FF6FC3A0000-00007FF6FC835000 DDNet.exe
00007FFC6B0F0000-00007FFC6B2E8000 ntdll.dll   	6.2.19041.2788
00007FFC6AB70000-00007FFC6AC2F000 KERNEL32.DLL	6.2.19041.2913
00007FFC68CE0000-00007FFC68FD6000 KERNELBASE.dll	6.2.19041.2913
00007FFC6AD40000-00007FFC6ADEF000 ADVAPI32.dll	6.2.19041.2913
00007FFC6A600000-00007FFC6A69E000 msvcrt.dll  	7.0.19041.546
00007FFC6A490000-00007FFC6A52C000 sechost.dll 	6.2.19041.2913
00007FFC6A6A0000-00007FFC6A7C6000 RPCRT4.dll  	6.2.19041.2965
00007FFC50410000-00007FFC5049A000 libcurl.dll 	8.0.1.0
00007FFC4F620000-00007FFC4F975000 discord_game_sdk.dll
00007FFC40CD0000-00007FFC41093000 avcodec-60.dll	60.3.100.0
00007FFC690E0000-00007FFC69107000 bcrypt.dll  	6.2.19041.2486
00007FFC6ADF0000-00007FFC6AE20000 IMM32.dll   	6.2.19041.2673
00007FFC69340000-00007FFC693AB000 WS2_32.dll  	6.2.19041.546
00007FFC6AA30000-00007FFC6AB5A000 ole32.dll   	6.2.19041.2965
00007FFC688C0000-00007FFC68A1E000 CRYPT32.dll 	6.2.19041.2965
00007FFC69BA0000-00007FFC69D3D000 USER32.dll  	6.2.19041.2788
00007FFC68FE0000-00007FFC690E0000 ucrtbase.dll	6.2.19041.789
00007FFC69D40000-00007FFC6A484000 SHELL32.dll 	6.2.19041.2788
00007FFC69160000-00007FFC69182000 win32u.dll  	6.2.19041.2913
00007FFC6AE20000-00007FFC6AE4C000 GDI32.dll   	6.2.19041.2913
00007FFC68A20000-00007FFC68ABD000 msvcp_win.dll	6.2.19041.789
00007FFC69840000-00007FFC69B94000 combase.dll 	6.2.19041.2788
00007FFC68AC0000-00007FFC68BD5000 gdi32full.dll	6.2.19041.2913
00007FFC50390000-00007FFC5040B000 avformat-60.dll	60.3.100.0
00007FFC6A7D0000-00007FFC6A825000 SHLWAPI.dll 	6.2.19041.2075
00007FFC40120000-00007FFC40324000 avutil-58.dll	58.2.100.0
00007FFC656F0000-00007FFC65705000 libopusfile.dll
00007FFC632B0000-00007FFC632D8000 swresample-4.dll	4.10.100.0
00007FFC4F570000-00007FFC4F61E000 libfreetype.dll	2.13.0.0
00007FFC4FC80000-00007FFC4FD27000 swscale-7.dll	7.1.100.0
00007FFC50680000-00007FFC506CD000 libpng16-16.dll
00007FFC3F910000-00007FFC3FB09000 SDL2.dll    	2.26.5.0
00007FFC6A530000-00007FFC6A5FD000 OLEAUT32.dll	6.2.19041.985
00007FFC693C0000-00007FFC69828000 SETUPAPI.dll	6.2.19041.2193
00007FFC4D970000-00007FFC4DAD4000 sqlite3.dll
00007FFC45890000-00007FFC459F4000 vulkan-1.dll	1.3.216.0
00007FFC69110000-00007FFC6915E000 cfgmgr32.dll	6.2.19041.1620
00007FFC64460000-00007FFC6446B000 ktmw32.dll  	6.2.19041.868
0000000064940000-0000000064955000 libwinpthread-1.dll	1.0.0.0
0000000062E80000-0000000062EA6000 zlib1.dll   	1.2.11.0
00007FFC687C0000-00007FFC687EE000 USERENV.dll 	6.2.19041.572
00007FFC60FA0000-00007FFC60FAA000 VERSION.dll 	6.2.19041.546
00007FFC3F590000-00007FFC3F6B5000 OPENGL32.dll	6.2.19041.2193
00007FFC64450000-00007FFC6445E000 steam_api.dll
00007FFC63D40000-00007FFC63D52000 libogg.dll
00007FFC41100000-00007FFC41192000 libopus.dll
00007FFC5C450000-00007FFC5C477000 WINMM.dll   	6.2.19041.546
00007FFC5DDD0000-00007FFC5DDFC000 GLU32.dll   	6.2.19041.2193
00007FFC68270000-00007FFC6827C000 CRYPTBASE.DLL	6.2.19041.546
00007FFC67180000-00007FFC67273000 dxgi.dll    	6.2.19041.2311
00007FFC667A0000-00007FFC667B2000 kernel.appcore.dll	6.2.19041.546
00007FFC68C50000-00007FFC68CD2000 bcryptPrimitives.dll	6.2.19041.2486
00007FFC662D0000-00007FFC6636E000 uxtheme.dll 	6.2.19041.2193
00007FFC574E0000-00007FFC577AE000 Start10_64.dll	1.9.7.0
00007FFC6B000000-00007FFC6B0AD000 shcore.dll  	6.2.19041.1865
00007FFC505B0000-00007FFC505EF000 exchndl.dll 	0.9.8.0
00007FFC3F3D0000-00007FFC3F492000 mgwhelp.dll 	0.9.8.0
00007FFC3EB40000-00007FFC3ED33000 dbghelp.dll 	10.0.18362.1
00007FFC504C0000-00007FFC504EA000 dbgcore.DLL 	10.0.18362.1
00007FFC68280000-00007FFC68298000 CRYPTSP.dll 	6.2.19041.546
00007FFC678C0000-00007FFC678F4000 rsaenh.dll  	6.2.19041.1052
00007FFC669A0000-00007FFC67133000 windows.storage.dll	6.2.19041.2788
00007FFC68320000-00007FFC6834E000 Wldp.dll    	6.2.19041.2788
00007FFC68800000-00007FFC6881F000 profapi.dll 	6.2.19041.844
00007FFC50700000-00007FFC5070C000 secur32.dll 	6.2.19041.546
00007FFC68780000-00007FFC687B2000 SSPICLI.DLL 	6.2.19041.2130
00007FFC67D00000-00007FFC67D3C000 iphlpapi.dll	6.2.19041.2788
00007FFC60320000-00007FFC603CE000 mscms.dll   	6.2.19041.746
00007FFC60530000-00007FFC60541000 ColorAdapterClient.dll	6.2.19041.546
00007FFC5F780000-00007FFC5F7C3000 icm32.dll   	6.2.19041.546
00007FFC13330000-00007FFC164E9000 nvoglv64.dll	31.0.15.1694
00007FFC645A0000-00007FFC645B4000 WTSAPI32.dll	6.2.19041.546
00007FFC684B0000-00007FFC684C2000 msasn1.dll  	6.2.19041.2251
00007FFC60F40000-00007FFC60F71000 cryptnet.dll	6.2.19041.906
00007FFC606A0000-00007FFC607E8000 drvstore.dll	6.2.19041.2546
00007FFC68670000-00007FFC6869C000 devobj.dll  	6.2.19041.1620
00007FFC68BE0000-00007FFC68C47000 WINTRUST.dll	6.2.19041.2913
00007FFC679E0000-00007FFC67A13000 ntmarta.dll 	6.2.19041.546
00007FFC6AC90000-00007FFC6AD39000 clbcatq.dll 	2001.12.10941.16384
00007FFC653A0000-00007FFC654F4000 WinTypes.dll	6.2.19041.2788
00007FFC6A910000-00007FFC6AA24000 MSCTF.dll   	6.2.19041.2673
00007FFC5D0F0000-00007FFC5D1EA000 textinputframework.dll	6.2.19041.2913
00007FFC65B60000-00007FFC65EBE000 CoreUIComponents.dll	6.2.19041.546
00007FFC65EF0000-00007FFC65FE2000 CoreMessaging.dll	6.2.19041.2193
00007FFC5DD50000-00007FFC5DD8B000 dxcore.dll  	6.2.19041.546
00007FFC665C0000-00007FFC665EF000 dwmapi.dll  	6.2.19041.746
00007FFC69320000-00007FFC6933D000 imagehlp.dll	6.2.19041.1415
00007FFC67150000-00007FFC67173000 gpapi.dll   	6.2.19041.2846
00007FFC3EA90000-00007FFC3EB3E000 NvCameraAllowlisting64.dll	7.1.816.0
00007FFC5D460000-00007FFC5DBA8000 nvapi64.dll 	31.0.15.1694
00007FFC628E0000-00007FFC629E4000 AppXDeploymentClient.dll	6.2.19041.2788
00007FFC67E40000-00007FFC67E8B000 powrprof.dll	6.2.19041.546
00007FFC67D40000-00007FFC67D52000 UMPDC.dll
00007FFC67B60000-00007FFC67BC4000 WINSTA.dll  	6.2.19041.2075
00007FFC68080000-00007FFC680EA000 mswsock.dll 	6.2.19041.546
00007FFC67140000-00007FFC6714D000 hid.dll     	6.2.19041.546
00007FFC471F0000-00007FFC47235000 dinput8.dll 	6.2.19041.1
00007FFC5CF90000-00007FFC5D0E2000 inputhost.dll	6.2.19041.1741
00007FFC64200000-00007FFC642F6000 PROPSYS.dll 	7.0.19041.1741
00007FFC67D70000-00007FFC67E3B000 DNSAPI.dll  	6.2.19041.2546
00007FFC6AB60000-00007FFC6AB68000 NSI.dll     	6.2.19041.610
00007FFC60400000-00007FFC6040A000 rasadhlp.dll	6.2.19041.546
00007FFC5F7E0000-00007FFC5F862000 fwpuclnt.dll	6.2.19041.2913
00007FFC677E0000-00007FFC67879000 schannel.DLL	6.2.19041.2913
00007FFC4EBA0000-00007FFC4EBB6000 mskeyprotect.dll	6.2.19041.2788
00007FFC68350000-00007FFC6838B000 NTASN1.dll  	6.2.19041.546
00007FFC68390000-00007FFC683B8000 ncrypt.dll  	6.2.19041.2788
00007FFC4EC50000-00007FFC4EC76000 ncryptsslp.dll	6.2.19041.2846
00007FFC54220000-00007FFC542CC000 TextShaping.dll
00007FFC45810000-00007FFC45883000 symsrv.dll  	10.0.18362.1
00007FFC52D70000-00007FFC53249000 wininet.dll 	11.0.19041.2193
00007FFC58B40000-00007FFC58DF1000 iertutil.dll	11.0.19041.2965
00007FFC47C60000-00007FFC47C77000 ondemandconnroutehelper.dll	6.2.19041.2311
00007FFC62C70000-00007FFC62D7A000 winhttp.dll 	6.2.19041.2673
00007FFC62740000-00007FFC6274B000 WINNSI.DLL  	6.2.19041.546
00007FFC57860000-00007FFC57A4C000 urlmon.dll  	11.0.19041.2788
00007FFC577B0000-00007FFC577D8000 srvcli.dll  	6.2.19041.1645
00007FFC67D60000-00007FFC67D6C000 netutils.dll	6.2.19041.546

Windows 6.2.19041.2913
DrMingw 0.9.8
```

**Comment by Robyt3** [comment]
```
0x0000000140300f87: src/base/system.cpp:202
 (inlined by) src/base/system.cpp:195
0x00000001401b8ebb: src/base/system.cpp:197
0x000000014029c812: /usr/x86_64-w64-mingw32/include/c++/12.2.0/bits/stl_vector.h:988
 (inlined by) src/engine/client/text.cpp:267
0x000000014029ccd7: src/engine/client/text.cpp:1036
0x000000014006e9b7: /usr/x86_64-w64-mingw32/include/c++/12.2.0/bits/shared_ptr_base.h:1070
 (inlined by) /usr/x86_64-w64-mingw32/include/c++/12.2.0/bits/shared_ptr_base.h:1524
 (inlined by) /usr/x86_64-w64-mingw32/include/c++/12.2.0/bits/shared_ptr.h:175
 (inlined by) src/engine/textrender.h:223
 (inlined by) src/game/client/components/hud.cpp:1647
0x000000014006f08a: src/game/client/components/hud.cpp:1609
 (inlined by) src/game/client/components/hud.cpp:1521
 (inlined by) src/game/client/components/hud.cpp:1495
0x00000001400ddfe5: src/game/client/gameclient.cpp:645
0x000000014001fbd1: src/engine/client/client.cpp:1033
 (inlined by) src/engine/client/client.cpp:1209
0x0000000140029348: src/engine/client/client.cpp:3286
0x00000001402ff0ad: src/engine/client/client.cpp:4807
0x00000001400013b1: /build/mingw-w64-crt/src/mingw-w64-v10.0.0/mingw-w64-crt/crt/crtexe.c:323
0x00000001400014c6: /build/mingw-w64-crt/src/mingw-w64-v10.0.0/mingw-w64-crt/crt/crtexe.c:178
```

Was fixed by #6634.
